reference-target-explainer.md

```diff
@@ -0,0 +1,611 @@
+    </div>
+  </template>
+</description-with-tooltip>
+```
+
+### Interaction with other features
+
+#### Interaction with CSS Selectors
+
+The referenceTarget does not affect CSS selectors in any way. An ID selector will target the host element that has the matching `id` attribute, and _not_ its referenceTarget.
+
+#### Form-associated custom elements
+
+A [form-associated custom element](https://html.spec.whatwg.org/dev/custom-elements.html#form-associated-custom-element) supports being the target of a label's `for` attribute. But if the element has a Reference Target for the `for` attribute, then the label applies to the target instead. There are no other changes to the behavior of a form-associated custom element.
+
+#### Nesting inside `<label>`
+
+There is no special support for an element nested inside a label. The label must use the `for` attribute to work with `referenceTarget`. It is still ok to nest the custom element inside the label, but it must also use `for`.
+
+```html
+<script>
+  customElements.define(
+    "fancy-input",
+    class FancyInput extends HTMLElement {
+      constructor() {
+        super();
+        this.shadowRoot_ = this.attachShadow({ mode: "closed" });
+        this.shadowRoot_.innerHTML = `<input id="real-input" />`;
+        this.shadowRoot_.referenceTarget = "real-input";
+      }
+    }
+  );
+</script>
+
+<!-- ❌ This label isn't associated with anything because it doesn't use `for`, 
+     and fancy-input is not form-associated. -->
+<label>
+  Fancy Input
+  <fancy-input></fancy-input>
+</label>
+
+<!-- ✅ This label is applied to the inner `<input id="real-input" />`, which is 
+     fancy-input's referenceTarget. -->
+<label for="fancy-input">
+  Fancy Input
+  <fancy-input id="fancy-input"></fancy-input>
+</label>
+```
+
+#### Nesting inside `<form>`
+
+Reference target does not change the behavior of the host element when it is nested inside a form. It does _not_ implicitly associate the target element with the form if it is not a form-associated custom element.
+
+#### JavaScript attributes that reflect `Element` objects
+
+Some JavaScript attributes reflect HTML attributes as Element objects rather than ID strings. These include:
+
+- ARIAMixin attributes like `ariaActiveDescendantElement`
+- `HTMLButtonElement.popoverTargetElement`
+- `HTMLInputElement.form`
+- `HTMLInputElement.labels`
+- `HTMLLabelElement.control`
+- _(This list is not exhaustive)_
+
+These will _always_ refer to the **host** element that they're targeting, and _never_ the referenceTarget element directly. This behavior maintains the design that an [IDL attribute with type Element](https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#reflecting-content-attributes-in-idl-attributes:element) can only refer to an element that is a descendant of a [shadow-including ancestor](https://dom.spec.whatwg.org/#concept-shadow-including-ancestor) of the element hosting the attribute.
```

Comment:
I like this. It exposes the minimal API surface while also (potentially) allowing authors with full access to open shadow roots to traverse the DOM and build the equivalent of `composedPath`/`activeElement`-drilling/etc.